src/decoders/sndfile.cpp

```diff
@@ -148,6 +146,26 @@ SharedPtr<Decoder> SndFileDecoderFactory::createDecoder(UniquePtr<std::istream>
     SndfilePtr sndfile(sf_open_virtual(&vio, SFM_READ, &sndinfo, file.get()));
     if(!sndfile) return nullptr;
 
+    SF_CUES cues;
+    Vector<uint64_t> cue_points;
+    if(sf_command(sndfile.get(), SFC_GET_CUE, &cues, sizeof(cues)))
```

Comment:
It seems `SF_CUES` and `SFC_GET_CUE` isn't supported on certain older versions of libsndfile. That will need a check, perhaps defining them appropriately if they aren't already. Also I don't think it's necessary to use a (dynamic) vector since it always has only two elements.

Reply:
I changed the Vector to a std::pair for consistency (instead of a simple Array), however I am not sure how one should go with the SFC_GET_CUE... For now I added a `#ifdef` statement which should fix the compilation problem, however Alure will probably not load if compiled with a newer libsndfile to be used in a system with an older libsndfile (older than 1.0.27, I believe)

Reply:
> however I am not sure how one should go with the SFC_GET_CUE

As long as libsndfile keeps backwards compatibility, you should be able to simply define `SFC_GET_CUE` to the correct value, and define `SF_CUES` the same way newer sndfile.h does. In such a case, when run with a libsndfile that supports it, `sf_command` should recognize the `SFC_GET_CUE` command and fill the struct appropriately when possible, while older libs won't recognize `SFC_GET_CUE` and the call will fail regardless.